### PR TITLE
Added missing Required method on string type.

### PR DESCRIPTION
**Pull request by GooRiOn**
# Info
This Pull Request is related to issue no. #128 

It turned out that we had ``Required()`` method for string which was simply put inside Property extensions (since it referce to all reference types) but it didn't support string.Empty. Therefore after adding ``Required()`` for string I had to change two unit tests.

# Changes
- Added Required() to the string type
- Fixed two unit tests
- Rearranged code in strategies unittests to fir pur template

### CODE REVIEW

File: src/Valit/ValitRuleStringExtensions.cs [hunk 1]
```diff
@@ -19,5 +19,8 @@ public static class ValitRuleStringExtensions
 
         public static IValitRule<TObject, string> Email<TObject>(this IValitRule<TObject, string> rule) where TObject : class
             => rule.Matches(@"\A(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?)\Z");
+
+        public static IValitRule<TObject, string> Required<TObject>(this IValitRule<TObject, string> rule) where TObject : class
+            => rule.Satisfies(p => !string.IsNullOrEmpty(p));
```
Comment (timdeschryver) [inline src/Valit/ValitRuleStringExtensions.cs:24]: Should we also check on whitespace here?

Comment (GooRiOn) [reply src/Valit/ValitRuleStringExtensions.cs:24]: Every string extension has just null or empty. I'd stick with that also here

File: tests/Valit.Tests/String/String_Required_Tests.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            });
+            
+            exception.ShouldBeOfType(typeof(ValitException));
+        }
+
+
+        [Theory]
+        [InlineData(false, true)]
+        [InlineData(true, false)] 
+        public void Int64_Required_Returns_Proper_Results_For_Nullable_Value(bool useNullValue,  bool expected)
```
Comment (timdeschryver) [inline tests/Valit.Tests/String/String_Required_Tests.cs]: Should be `String_Required_xxx`

File: tests/Valit.Tests/String/String_Required_Tests.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        public String_Required_Tests()
+        {
+            _model = new Model();
+        }
+
+        private readonly Model _model;
+
+        class Model
+        {
+            public string EmptyValue => string.Empty;
```
Comment (timdeschryver) [inline tests/Valit.Tests/String/String_Required_Tests.cs:70]: No tests are written for the `EmptyValue`

**Review by timdeschryver — CHANGES_REQUESTED** [review]
LGTM, just some nitpicking.

**Comment by GooRiOn** [comment]
fixed

File: tests/Valit.Tests/String/String_Required_Tests.cs [hunk 4]
```diff
@@ -18,19 +18,43 @@ public void String_Required_For_Nullable_Value_Throws_When_Null_Rule_Is_Given()
...
+
+            Assert.False(result.Succeeded);
+        }
+
+        [Fact]
+        public void String_Required_Fails_For_Empty_Value()
+        {            
+            IValitResult result = ValitRules<Model>
+                .Create()
+                .Ensure(m => m.NullValue, _=>_
```
Comment (timdeschryver) [inline tests/Valit.Tests/String/String_Required_Tests.cs]: Should be `EmptyValue` here 😉

Comment (GooRiOn) [reply tests/Valit.Tests/String/String_Required_Tests.cs]: what's wrong with me :D

**Review by timdeschryver — APPROVED** [review]

**Pull request merged**